### ISSUE TITLE
ValueError: setting an array element with a sequence.

### ISSUE THREAD

**Issue by nicolas-gervais** [naiveHobo/InvoiceNet#16]
I pre-processed all the data as indicated and I got this:
```
Exception in thread Thread-107:
Traceback (most recent call last):
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\threading.py", line 926, in _bootstrap_inner
    self.run()
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\nicol\Documents\InvoiceNet\invoicenet\gui\trainer.py", line 253, in _train
    train_loss = model.train_step(next(train_iter))
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\eager\def_function.py", line 780, in __call__
    result = self._call(*args, **kwds)
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\eager\def_function.py", line 823, in _call
    self._initialize(args, kwds, add_initializers_to=initializers)
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\eager\def_function.py", line 697, in _initialize
    *args, **kwds))
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\eager\function.py", line 2855, in _get_concrete_function_internal_garbage_collected
    graph_function, _, _ = self._maybe_define_function(args, kwargs)
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\eager\function.py", line 3213, in _maybe_define_function
    graph_function = self._create_graph_function(args, kwargs)
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\eager\function.py", line 3075, in _create_graph_function
    capture_by_value=self._capture_by_value),
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\framework\func_graph.py", line 986, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\eager\def_function.py", line 600, in wrapped_fn
    return weak_wrapped_fn().__wrapped__(*args, **kwds)
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\eager\function.py", line 3735, in bound_method_wrapper
    return wrapped_fn(*args, **kwargs)
  File "C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\framework\func_graph.py", line 973, in wrapper
    raise e.ag_error_metadata.to_exception(e)
ValueError: in user code:

    C:\Users\nicol\Documents\InvoiceNet\invoicenet\acp\acp.py:85 train_step  *
        predictions = self.model(inputs, training=True)
    C:\Users\nicol\Documents\InvoiceNet\invoicenet\acp\model.py:176 call  *
        parsed = self.parser(inputs=(x, context), training=training)
    C:\Users\nicol\Documents\InvoiceNet\invoicenet\parsing\parsers.py:65 call  *
        empty_answer = tf.constant(InvoiceData.eos_idx, tf.int32, shape=(tf.shape(x)[0], self.seq_out))
    C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\framework\constant_op.py:264 constant  **
        allow_broadcast=True)
    C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\framework\constant_op.py:282 _constant_impl
        allow_broadcast=allow_broadcast))
    C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\tensorflow\python\framework\tensor_util.py:453 make_tensor_proto
        if shape is not None and np.prod(shape, dtype=np.int64) == 0:
    <__array_function__ internals>:6 prod

    C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\numpy\core\fromnumeric.py:2962 prod
        keepdims=keepdims, initial=initial, where=where)
    C:\Users\nicol\anaconda3\envs\invoicenet\lib\site-packages\numpy\core\fromnumeric.py:90 _wrapreduction
        return ufunc.reduce(obj, axis, dtype, out, **passkwargs)

    ValueError: setting an array element with a sequence.
```

**Comment by brandomr** [comment]
I'm not 100% sure, but I think this may be due to missing data in the training set. Do you know if you should leave out keys that don't exist (optional fields) or should they be included in the training `JSON` but set to `None` or `null`?

**Comment by naiveHobo** [comment]
I don't think that's the issue, if the training field is not available in any label, it is automatically assumed to be null. This actually turned out to be an error in the optional parser (sorry about that). Should be fixed in fe6ad79d53b5cae08dd6383c912543f5b5fb9950.

**Comment by brandomr** [comment]
@naiveHobo awesome, I'll give it a shot asap and let you know if that resolves things.
